### PR TITLE
NMS-14118: Add SSH host key verification

### PR DESCRIPTION
**Pull request by fooker**
### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/14118

### CODE REVIEW

**Review by Bonrob2 — APPROVED** [review]
Docs LGTM!

File: features/device-config/ssh-scripting/src/main/java/org/opennms/features/deviceconfig/sshscripting/impl/SshScriptingServiceImpl.java [hunk 1]
```diff
@@ -138,18 +140,33 @@ private static class SshInteractionImpl implements SshInteraction, AutoCloseable
...
-                int port,
+                final SocketAddress target,
+                final String hostKeyFingerprint,
                 Map<String, String> vars,
                 Duration timeout,
                 InetAddress tftpServerIPv4Address,
                 InetAddress tftpServerIPv6Address
         ) throws Exception {
             sshClient = SshClient.setUpDefaultClient();
+            sshClient.setServerKeyVerifier((clientSession, socketAddress, publicKey) -> {
```
Comment (cgorantla) [inline features/device-config/ssh-scripting/src/main/java/org/opennms/features/deviceconfig/sshscripting/impl/SshScriptingServiceImpl.java:151]: Can we add a test to verify this path

**Review by cgorantla — COMMENTED** [review]
LGTM! 
Will be good to have test coverage for this addition!

File: features/device-config/ssh-scripting/src/main/java/org/opennms/features/deviceconfig/sshscripting/impl/SshScriptingServiceImpl.java (hunk 1 above)
Comment (fooker) [reply features/device-config/ssh-scripting/src/main/java/org/opennms/features/deviceconfig/sshscripting/impl/SshScriptingServiceImpl.java:151]: test altered to cover this.

**Review by cgorantla — APPROVED** [review]

**Pull request merged**